### PR TITLE
DEVOPS-1440 Setup Ruff

### PR DESCRIPTION
**Pull request by bio-boris**
* Previous [PR](https://github.com/kbase/staging_service/pull/195) had too much going on, and had changes from black that didn't line up
* Added precommit
* Added gha to check linting
* Added minimal fixes

### CODE REVIEW

File: import_specifications/clients/baseclient.py [hunk 1]
```diff
@@ -51,8 +51,7 @@ def _get_token(user_id, password, auth_svc):
         tok = _json.loads(ret.text)
     elif status == 403:
         raise Exception(
-            "Authentication failed: Bad user_id/password "
-            + "combination for user %s" % (user_id)
+            "Authentication failed: Bad user_id/password " + "combination for user %s" % (user_id)
```
Comment (ialarmedalien) [inline import_specifications/clients/baseclient.py:54]: if you write this as
```py
f"Authentication failed: Bad user_id/password combination for user {user_id}"
```
does it break the line length rule?

File: tests/import_specifications/test_file_writers.py [hunk 2]
```diff
@@ -239,10 +239,7 @@ def test_file_writers_fail():
                 "data": [],
             }
         },
-        E(
-            "Invalid order_and_display entry for datatype t at index 2 - "
-            + "expected 2 item list"
-        ),
+        E("Invalid order_and_display entry for datatype t at index 2 - " + "expected 2 item list"),
```
Comment (ialarmedalien) [inline tests/import_specifications/test_file_writers.py:242]: is
```py
E("Invalid order_and_display entry for datatype t at index 2 - expected 2 item list"),
```
too long?

File: tests/import_specifications/test_file_writers.py [hunk 3]
```diff
@@ -252,10 +249,7 @@ def test_file_writers_fail():
                 "data": [],
             }
         },
-        E(
-            "Invalid order_and_display entry for datatype t at index 0 - "
-            + "expected 2 item list"
-        ),
+        E("Invalid order_and_display entry for datatype t at index 0 - " + "expected 2 item list"),
```
Comment (ialarmedalien) [inline tests/import_specifications/test_file_writers.py:252]: same question here

File: tests/import_specifications/test_file_writers.py [hunk 4]
```diff
@@ -317,10 +311,7 @@ def test_file_writers_fail():
                 "data": [{"foo": 2, "whee": 3}, {"foo": 1, "whee": []}],
             }
         },
-        E(
-            "Data type ty data row 1's value for parameter whee "
-            + "is not a number or a string"
-        ),
+        E("Data type ty data row 1's value for parameter whee " + "is not a number or a string"),
```
Comment (ialarmedalien) [inline tests/import_specifications/test_file_writers.py:314]: and here

File: import_specifications/clients/baseclient.py (hunk 1 above)
Comment (bio-boris) [reply import_specifications/clients/baseclient.py:54]: I have not checked, but I would prefer to leave this autogenerated file untouched/ignored.

**Review by ialarmedalien — APPROVED** [review]

**Pull request merged**